### ISSUE TITLE
Idea: Adjust window split according to the window minimum size

### ISSUE THREAD

**Issue by Chrysostomus** [pop-os/shell#85]
Currently, when a new window is spawned in a split that is smaller than it's minimum size, the new window overlaps with the existing windows. I'm not sure if it is at all feasible, but it would look really good if instead of splitting the focused window 50/50, the split was adjusted to accomodate for the minimum size of the new window. It would also minimize the need of manual resizing on a smaller monitor.

So instead of this:

![Screenshot from 2020-03-17 23-56-59](https://user-images.githubusercontent.com/12002226/76905678-1046e780-68ab-11ea-8c81-1e1fefe61847.png)
There would be this:

![Screenshot from 2020-03-17 23-57-10](https://user-images.githubusercontent.com/12002226/76905694-189f2280-68ab-11ea-89c1-52c7e6594a32.png)

You could calculate the split from the size of the current window - the gap - program specified minimum size of the new window. Since the windows are spawned before being placed, the data should be available.

**Comment by timrichardson** [comment]
If I have two Chromium windows tiled on my middle monitor and adjust the right side until it will shink no more, and then if I expand the larger left side, the right side does not shrink to make room ... it moves rightwards,which mean it move partly on to the adjoining monitor. In the minimum size is to be respected, the left side should have a maximum size.

**Comment by Holzhaus** [comment]
The alternative would be to force-resize windows. i3 can do this, and currently using Steam on a 1920x1080p screen does not work well with tiling in pop shell.

**Comment by Chrysostomus** [comment]
Not quite as awesome but probably a lot easier to do it that way.

**Comment by Holzhaus** [comment]
Yes, especially if the window overlaps both vertically and horizontally:

![Screenshot from 2020-09-06 15-34-52](https://user-images.githubusercontent.com/1834516/92327024-f6ff4680-f056-11ea-9f13-04592057fd46.png)

**Comment by RickDW** [comment]
If it's at all possible I second Holzhaus's  forced resizing. For some windows it makes sense to have a restricted size, but as far as I've seen it's not nearly always required. In my opinion it should be up to the user to decide what size your windows are, and if the required size is actually better you can always resize it like that yourself.

**Comment by GaryBer** [comment]
Steam has a minimum window width which causes these overlaps :/

![image](https://user-images.githubusercontent.com/34002641/97183569-f0926e80-176b-11eb-9563-874c5184fc15.png)
